app/controllers/application_controller/ci_processing.rb

```diff
@@ -45,36 +45,21 @@ def edit_record
       @redirect_controller = "host"
       session[:host_items] = obj if obj.length > 1
     end
-    @redirect_id = obj[0] if obj.length == 1      # not redirecting to an id if multi host are selected for credential edit
-
-    if !["ScanItemSet", "Condition", "Schedule", "MiqAeInstance"].include?(db)
-      @refresh_partial = "edit"
-      @refresh_partial = "edit_set" if params[:db] == "policyprofile"
-    else
-      if db == "ScanItemSet"
-        scan = ScanItemSet.find(obj[0])
-        if !scan.read_only
-          @refresh_partial = "edit"
-        else
-          @refresh_partial = "show_list_set"
-        end
-      elsif db == "Condition"
-        cond = Condition.find(obj[0])
-        if cond.filename.nil?
-          @refresh_partial = "edit"
-        else
-          @refresh_partial = "show_list"
-        end
-      elsif db == "Schedule" && params[:controller] != "report"
-        sched = MiqSchedule.find(obj[0])
-        @refresh_partial = "edit"
-      elsif db == "Schedule" && params[:controller] == "report"
-        sched = MiqSchedule.find(obj[0])
-        @refresh_partial = "schedule_edit"
-      elsif db == "MiqAeInstance"
-        @refresh_partial = "instance_edit"
-      end
-    end
+    @redirect_id = obj[0] if obj.length == 1 # not redirecting to an id if multi host are selected for credential edit
+
+    @refresh_partial = if !%w(ScanItemSet Condition Schedule MiqAeInstance).include?(db)
```

Comment:
Maybe something as following?
```
case db
when "ScanItemSet"
  ScanItemSet.find(obj[0]).read_only ? 'show_list_set' : 'edit'
...
when "Schedule"
  params[:controller] == 'report' ? 'schedule_edit' : 'edit'
else
  params[:db] == "policyprofile" ? 'edit_set' : 'edit'
end
```

Reply:
Sure, what you're suggesting looks better. Fixed.